### DISCUSSION TITLE
websocket connection to 'ws:// failed

### DISCUSSION THREAD

**Discussion by JproChen** [miguelgrinberg/Flask-SocketIO#1973] [Q&A] [answered]
Well my code was working fine before
but like a week ago, the chrome has the error
websocket connection to 'ws:// failed
and then the socketio just transfer to use polling
I am using gevent

from gevent import monkey
monkey.patch_all(ssl=False)
...
socketio.run(app, host='172.18.9.122', port=8080)
and in js
var socket = io.connect('http://172.18.9.122:8080/');
Is there and policy update of chrome causes this problem?

**Comment by miguelgrinberg** [comment]
Did you install gevent-websocket?

**Reply by JproChen** [reply]
Yes, I did install gevent-websocket
And here is my server log
code 400, message Bad request version ('::\x13\x03\x13\x01\x13\x02Ì©Ì¨À+À/À,À0À\x13À\x14\x00\x9c\x00\x9d\x00/\x005\x01\x00\x01\x93ÊÊ\x00\x00\x00\x05\x00\x05\x01\x00\x00\x00\x00ÿ\x01\x00\x01\x00\x00\x1b\x00\x03\x02\x00\x02\x00')
192.168.168.23 - - [18/Apr/2023 18:21:29] "üÉK§Ó3ª]ÅHÖ¿¦ÿ}âè0 m~CWÈÜ34 Vá×Xr`Ò®¥&åv{w¸Pâ÷º@X3~ ::Ì©Ì¨À+À/À,À0ÀÀÊÊÿ00 -

Is this caused by cross origin?
but I'm certain that I haven't change any code and package till this problem occur

**Reply by miguelgrinberg** [reply]
Not 100% sure, but this could be that the client is attempting to connect with encryption, but your server does not have TLS configured.

**Reply by JproChen** [reply, marked as answer]
It seems like my python script didn't ended correctly
I thought I was using gevent, but somehow it switched to eventlet
and when I force it to use gevent, it says the port was already in use.
killing the process in the port, then restart the script solved my problem, thanks for your time.
